rclcpp/src/rclcpp/exceptions.cpp

```diff
@@ -73,6 +73,40 @@ throw_from_rcl_error(
   }
 }
 
+  std::exception_ptr
+  from_rcl_error(
+    rcl_ret_t ret,
+    const std::string & prefix,
+    const rcl_error_state_t * error_state,
+    void (* reset_error)())
+  {
+    if (RCL_RET_OK == ret) {
+      return std::make_exception_ptr(std::invalid_argument("ret is RCL_RET_OK"));
```

Comment:
This exception is different since it is a problem with the arguments passed to `from_rcl_error` instead of an exception created to match `ret`. Recommend `throw std::invalid_argument("...")` instead.